### ISSUE TITLE
Question about the query ranking.

### ISSUE THREAD

**Issue by pkqbajng** [yuxumin/PoinTr#137]
Hi, sir, thanks for your excellent job. Present implementation achieve the Adaptive Query Generation using torch.argsort and torch.gather. However, this will leading to that the parameters of the query ranking module won't update during training. You mentioned that this can be solved using gumble softmax, could you please share the code about this part?

**Comment by Yiju1213** [comment]
The author said that they already use the gumble-softmax operation, but also I can't find any implementation.

**Comment by Yiju1213** [comment]
Still it's excellent job and I'm currently using it. It would just be perfect if this part could be updated!
